src/idl_gen_csharp.cpp

```diff
@@ -934,10 +934,15 @@ class CSharpGenerator : public BaseGenerator {
               auto qualified_name = NamespacedName(sd);
               code += "  public " + qualified_name + "? ";
               code += Name(field) + "ByKey(";
-              code += GenTypeGet(key_field.value.type) + " key)";
+              code += GenTypeGet(key_field.value.type) + " key";
+              if (IsString(key_field.value.type)) {
+                code += ", byte[] tmpBuffer = null";
```

Comment:
tmpBuffer isn't a good name since I don't know what it is supposed to store by just reading it.

Reply:
I've changed this to `keyBuffer`